PYIC-8800: Update F2F error routing for mitigation paths

AmritSidhu opened the pull request:
## Proposed changes
### What changed

- Update F2F error routing for mitigation paths
- Route V03 F2F errors via sorryTechnicalErrorEnabled mitigation flow

### Why did it change

- We have new error screens and routing for handling unexpected outages for the Passport, Driving Licence, KBV and F2F CRIs. However, the current routing allows for users to go down unnecessary routes when on a mitigation journey.

For example, this user flow:
User starts IPV through the web route via Passport CRI → gets a ...
User redirected to DL CRI → CRI server_errors
User shown new sorry-technical-problem page - shown options to go through the app or post office

Neither of these options allow mitigating the ... from the passport CRI and so will fail regardless of the documents the user re-submits, sending them down unnecessary journeys.

### Issue tracking
<!-- Jira ticket & other docs, like RFCs -->

- [PYIC-8800](https://govukverify.atlassian.net/browse/PYIC-8800)

## Checklists

- [ ] READMEs and documentation up-to-date
- [ ] API/ unit/ contract tests have been written/ updated
- [ ] No risk of exposure: PII, credentials, etc through logs/ code
- [ ] Production changes appropriately staged out


[PYIC-8800]: https://govukverify.atlassian.net/browse/PYIC-8800?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

AmritSidhu committed:
PYIC-8800: Update F2F error routing for mitigation paths

Route mitigation F2F errors via sorryTechnicalErrorEnabled mitigation flow

thebauSoftwire commented on lambdas/process-journey-event/src/main/resources/statemachine/journey-maps/f2f-hand-off.yaml:
I can't comment on line 106 directly but we want to route to a slightly different variant of the `sorry-technical-problem` page which doesn't give the option to back through the web route ([this page](https://identity.build.account.gov.uk/dev/template/sorry-technical-problem/en?context=kbvMitigation)) since they got a CI through the web route

thebauSoftwire commented on lambdas/process-journey-event/src/main/resources/statemachine/journey-maps/f2f-hand-off.yaml:
```diff
@@ -117,8 +117,18 @@ states:
+          enhanced-verification:
+            targetJourney: NEW_P2_IDENTITY
+            targetState: APP_DOC_CHECK_PYI_ESCAPE
+            targetEntryEvent: next
```
conditional check makes sense :) we should also nest a "checkJourneyContext" (similar to line 146) so users are routed back to the correct LoC identity journey

thebauSoftwire replied:
sorry ignore the above, since the checkMitigation will be done on line 106 to route to the mitigation page variant, we don't need to check it on the page state. We just need to check for the journey context to route the user to the correct identity journey

AmritSidhu committed:
PYIC-8800: Route to mitigation variant of the error page

Check journey context to route the user to correct journey

AmritSidhu committed:
PYIC-8800: Fix routing and add API coverage for F2F mitigation retry and app flows

fix targetState when wanting to go back through app after server_error on f2f
correct syntax mistakes

thebauSoftwire commented on api-tests/features/unexpected-cri-error.feature:
```diff
@@ -516,3 +516,79 @@ Feature: Handling unexpected CRI errors
+      When I poll for async DCMAW credential receipt
+      Then the poll returns a '201'
+      When I submit the returned journey event
+      Then I get a 'page-dcmaw-success' page response
```
just a small thing but I think it might be worth adding additional steps to this test to check we get a P2 at the end of it. Just so we can make sure that we're routing correctly back to the medium-confidence journey

thebauSoftwire commented on api-tests/features/unexpected-cri-error.feature:
```diff
@@ -516,3 +516,79 @@ Feature: Handling unexpected CRI errors
+      When I poll for async DCMAW credential receipt
+      Then the poll returns a '201'
+      When I submit the returned journey event
+      Then I get a 'page-dcmaw-success' page response
```
I don't think this is testing the mitigation via f2f cri then unexpected error route. Should this instead be:
```
     # Return journey
      When I start new 'medium-confidence' journeys until I get a 'page-ipv-identity-document-start' page response
      And I submit an 'end' event
      Then I get a 'page-ipv-identity-postoffice-start' page response
      When I submit a 'next' event
      Then I get a 'claimedIdentity' CRI response
      When I submit 'kenneth-current' details to the CRI stub
      Then I get an 'address' CRI response
      When I submit 'kenneth-current' details to the CRI stub
      Then I get a 'fraud' CRI response
      When I submit 'kenneth-score-2' details with attributes to the CRI stub
        | Attribute          | Values                   |
        | evidence_requested | {"identityFraudScore":2} |
      Then I get an 'f2f' CRI response
      Then I get a 'f2f' CRI response
      When I call the CRI stub with attributes and get a 'server_error' OAuth error
        | Attribute          | Values                                          |
        | evidence_requested | {"scoringPolicy":"gpg45","strengthScore":0} |
      Then I get a 'sorry-technical-problem' page response with context 'kbvMitigation'
      When I submit a 'app' event
      Then I get a 'page-ipv-identity-document-start' page response
      When I submit a 'returnToRp' event <--- testing the return to RP route just to cover that case
      Then I get an OAuth response
      When I use the OAuth response to get my identity
      Then I get a 'P0' identity
```

thebauSoftwire replied:
sorry, missed this in the first review!

thebauSoftwire committed:
PYIC-8800: skip address + fraud CRIs when a user drops out of f2f cri during a KBV mitigation due to an unexpected error

We shouldn't allow users to go through the fraud and address again CRIs during same-session and separate-session KBV mitigations after an unexpected error from the f2f CRI as they have already gone through it prior to that.

thebauSoftwire commented on lambdas/process-journey-event/src/main/resources/statemachine/journey-maps/f2f-hand-off.yaml:
```diff
@@ -123,6 +126,27 @@ states:
+        checkJourneyContext:
+          mediumConfidence:
+            targetJourney: NEW_P2_IDENTITY
+            targetState: KBV_MITIGATION_VIA_APP_SKIP_ADDRESS_FRAUD
```
In both separate-session and same-session KBV mitigations, the user would already have been through the address and fraud CRIs so we want to skip them if the users decides to go through the app again
